src/full_node/blockchain.py

```diff
@@ -149,6 +152,19 @@ async def _load_chain_from_store(self,) -> None:
         while True:
             self.headers[cur_b.header_hash] = cur_b
             self.height_to_hash[cur_b.height] = cur_b.header_hash
+            full_block: Optional[FullBlock] = await self.block_store.get_block(
```

Comment:
This is not efficient. It is reading and parsing every single block in the blockchain, only to obtain the challenge hash and iters. Much better would be to store challenge hash and iters next to the block header, or in a new table

Reply:
Actually challenge hash and iterations are already stored in the header right? So why do we need to get_block

Reply:
Also, this while loop does not include every single block. Between the LCA and the tips, there are a few more blocks added before this loop (line 141). These need to be included in the pot_to_chain_height dict as well.

Reply:
I think PoT belongs to HeaderBlock type, but we have only Header type stored here. I think I'll go for another table to store (challenge_hash, iter, height), and update it on every add_block call.

Reply:
Wait a second, the header has challenge hash and total_iters in it. You can subtract the iters of the previous block to get the iters, you don't need proof of time.